src/Raven.Server/Commercial/LicenseManager.cs

```diff
@@ -235,65 +383,77 @@ private void ReloadLicense()
                 {
-                    await HandleLeaseLicenseFailure(response).ConfigureAwait(false);
-                    return;
+                    var severity =
+                        leasedLicense.NotificationSeverity == NotificationSeverity.None ? 
+                        NotificationSeverity.Info : leasedLicense.NotificationSeverity;
+                    var alert = AlertRaised.Create(
+                        "License was updated",
+                        hasMessage ? leasedLicense.Message : "license was updated",
+                        AlertType.LicenseManager_LicenseUpdateMessage,
+                        severity);
+
+                    _serverStore.NotificationCenter.Add(alert);
                 }
 
-                var licenseAsStream = await response.Content.ReadAsStreamAsync().ConfigureAwait(false);
-                using (var context = JsonOperationContext.ShortTermSingleUse())
-                {
-                    var json = context.Read(licenseAsStream, "leased license");
-                    var newLicense = JsonDeserializationServer.License(json);
-                    if (newLicense.Name == license.Name && newLicense.Id == license.Id &&
-                        newLicense.Keys.SequenceEqual(license.Keys))
-                        return;
+                return licenseChanged ? leasedLicense.License : null;
+            }
+        }
 
-                    await Activate(newLicense, skipLeaseLicense: true);
-                }
+        private async Task LeaseLicense()
+        {
+            if (_serverStore.IsLeader() == false)
+            {
+                // only the leader is in charge of updating the license
+                return;
+            }
 
-                var message = "License was updated";
-                if (_licenseStatus.Attributes.TryGetValue("expiration", out object expiration) && expiration is DateTime)
-                {
-                    var expirationDate = ((DateTime)expiration).ToString("yyyy-MMM-dd", CultureInfo.CurrentUICulture);
-                    message += $", new expiration date is {expirationDate}";
-                }
+            if (_leaseLicenseSemaphore.Wait(0) == false)
+                return;
 
-                var alert = AlertRaised.Create(
-                    "License updated",
-                    message,
-                    AlertType.LicenseManager_LicenseUpdated,
-                    NotificationSeverity.Info);
+            try
+            {
+                var loadedLicense = _serverStore.LoadLicense();
+                if (loadedLicense == null)
+                    return;
 
-                _serverStore.NotificationCenter.Add(alert);
+                var updatedLicense = await GetUpdatedLicense(loadedLicense);
+                if (updatedLicense == null)
+                    return;
+
+                await Activate(updatedLicense, skipLeaseLicense: true);
             }
             catch (Exception e)
             {
```

Comment:
Here, change "could not lease license" to "could not validate license"

Reply:
Need a configuration option that would disable remote license check for closed environments where they wouldn't want to get this error